modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/util/BSTValidator.java

```diff
@@ -75,12 +76,11 @@ public Credential validate(Credential credential, RequestData requestData) throw
     public boolean authenticate(String binarySecurityToken) throws
             AuthenticationException {
 
-        if("123456789123456789".equals(binarySecurityToken)){
-            System.out.println("TRUE");
+        CacheEntry cacheentry = (CacheEntry) DeviceUtil.getCacheEntry(binarySecurityToken);
+        String username = cacheentry.getUsername();
+        if (username != null) {
             return true;
-        }
-        else {
-            System.out.println("FALSE");
+        } else {
             return false;
         }
```

Comment:
Here you can simply do "return (username != null)" without any if else conditions.